app/controllers/bus_stops_controller.rb

```diff
@@ -21,6 +22,7 @@ def by_sequence
   end
 
   def by_status
+    session.delete(:route_id)
```

Comment:
Why bother, out of curiosity (here as well as below)? If it only matters that it's there when we need it, does it matter if it's there when we don't?

Reply:
See my general comment - I would be interested in determining whether sessions are actually necessary for this feature, and then returning to the best way to engineer it.